unstructured/ingest/cli/cmds/azure_cognitive_search.py

```diff
@@ -0,0 +1,92 @@
+import logging
+from dataclasses import dataclass
+
+import click
+
+from unstructured.ingest.cli.cmds.utils import conform_click_options
+from unstructured.ingest.cli.common import (
+    log_options,
+)
+from unstructured.ingest.cli.interfaces import (
+    CliMixin,
+    CliPartitionConfig,
+    CliReadConfig,
+)
+from unstructured.ingest.interfaces import BaseConfig
+from unstructured.ingest.logger import ingest_log_streaming_init, logger
+from unstructured.ingest.runner import runner_map
+
+
+@dataclass
+class AzureCognitiveSearchCliWriteConfig(BaseConfig, CliMixin):
+    key: str
+    endpoint: str
+    index: str
+
+    @staticmethod
+    def add_cli_options(cmd: click.Command) -> None:
+        options = [
+            click.Option(
+                ["--key"],
+                required=True,
+                type=str,
+                help="Key credential used for authenticating to an Azure service.",
+                envvar="AZURE_SEARCH_API_KEY",
+                show_envvar=True,
+            ),
+            click.Option(
+                ["--endpoint"],
+                required=True,
+                type=str,
+                help="The URL endpoint of an Azure search service. "
+                "In the form of https://{{service_name}}.search.windows.net",
+                envvar="AZURE_SEARCH_ENDPOINT",
+                show_envvar=True,
+            ),
+            click.Option(
+                ["--index"],
+                required=True,
+                type=str,
+                help="The name of the index to connect to",
+            ),
+        ]
+        cmd.params.extend(options)
+
+
+@click.command(name="azure-cognitive-search")
+@click.pass_context
+def azure_cognitive_search_dest(ctx: click.Context, **options):
+    if not ctx.parent:
+        raise click.ClickException("destination command called without a parent")
+    if not ctx.parent.info_name:
+        raise click.ClickException("parent command missing info name")
+    source_cmd = ctx.parent.info_name.replace("-", "_")
+    runner_fn = runner_map[source_cmd]
+    parent_options: dict = ctx.parent.params if ctx.parent else {}
+    conform_click_options(options)
+    conform_click_options(parent_options)
+    verbose = parent_options.get("verbose", False)
+    ingest_log_streaming_init(logging.DEBUG if verbose else logging.INFO)
+    log_options(parent_options, verbose=verbose)
+    log_options(options, verbose=verbose)
```

Comment:
not for this pr, but at some point maybe we should consider if there's anything we can do to reduce the duplicated steps across all of these?